app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,67 @@
+class VendorsController < ApplicationController
+	def index
+		session[:user_type] = params[:user_type]
+	  @vendors = Market.find(params[:market_id]).vendors
+		@vendors.sort_by! {|vendor| vendor[:name]}
+	end
+
+	def all
+		@vendors = Vendor.all.order(:name)
+	end
+
+	def new
+	  @vendor = Vendor.new
+	end
+
+	def create
+	  vendor = Vendor.create(vendor_params[:vendor])
+	  vendor.market_id = params[:market_id]
+	  vendor.save
+	  @buttons = true
+	  redirect_to market_path(id: vendor.market_id)
+	end
+
+	def edit
+		@vendor = Vendor.find(params[:id])
+	end
+
+	def show
+		@vendor = Vendor.find(params[:id])
+		@sales = []
+		@vendor.products.each do |product|
+			product.sales.each do |sale|
+				@sales << sale
+			end
+		end
+		@sales = @sales if params[:order].nil?
```

Comment:
I'm wondering if this will do anything. If the condition does not evaluate to true, won't `@sales` still be equal to itself?